lib/elixir/lib/module/types.ex

```diff
@@ -505,6 +505,10 @@ defmodule Module.Types do
   defp map_type?({:map, _}), do: true
   defp map_type?(_other), do: false
 
+  defp maplike_type?({:map, _}), do: true
+  defp maplike_type?({:union, union}), do: Enum.all?(union, &maplike_type?/1)
```

Comment:
Should this be `Enum.any?`? Any entry is enough to make it map like, right?

```suggestion
  defp maplike_type?({:union, union}), do: Enum.any?(union, &maplike_type?/1)
```